addons/nuxeo-retention/nuxeo-retention-core/src/main/java/org/nuxeo/retention/adapters/RetentionRule.java

```diff
@@ -0,0 +1,248 @@
+        if (propertyValue == null) {
+            return Collections.emptyList();
+        }
+        return Arrays.asList((String[]) propertyValue);
+    }
+
+    public String getExpression() {
+        return (String) document.getPropertyValue(RetentionConstants.EXPRESSION_PROP);
+    }
+
+    public String getMetadataXpath() {
+        return (String) document.getPropertyValue(RetentionConstants.METADATA_XPATH_PROP);
+    }
+
+    public Calendar getRetainUntilDateFrom(Calendar calendar) {
+        LocalDateTime datetime = LocalDateTime.ofInstant(calendar.getTime().toInstant(), ZoneId.systemDefault());
+        return getRetainUntilDateFrom(datetime);
+    }
+
+    protected Calendar getRetainUntilDateFrom(LocalDateTime datetime) {
+        LocalDateTime localDateTime = datetime.plusYears(getDurationYears())
+                                              .plusMonths(getDurationMonths())
+                                              .plusDays(getDurationDays())
+                                              .plus(getDurationMillis(), ChronoUnit.MILLIS);
+        return GregorianCalendar.from(localDateTime.atZone(ZoneId.systemDefault()));
+    }
+
+    public Calendar getRetainUntilDateFromNow() {
+        return getRetainUntilDateFrom(LocalDateTime.now());
+    }
+
+    public String getStartingPointEvent() {
+        return (String) document.getPropertyValue(RetentionConstants.STARTING_POINT_EVENT_PROP);
+    }
+
+    public String getStartingPointExpression() {
+        return (String) document.getPropertyValue(RetentionConstants.STARTING_POINT_EXPRESSION_PROP);
+    }
+
+    public StartingPointPolicy getStartingPointPolicy() {
+        String value = (String) document.getPropertyValue(RetentionConstants.STARTING_POINT_POLICY_PROP);
+        if (value != null) {
+            return StartingPointPolicy.valueOf(value.toUpperCase());
+        }
+        return null;
+
+    }
+
+    public boolean isAfterDelay() {
+        return StartingPointPolicy.AFTER_DELAY.equals(getStartingPointPolicy());
+    }
+
+    public boolean isAuto() {
+        return ApplicationPolicy.AUTO.name().toLowerCase().equals(getApplicationPolicy());
+    }
+
+    public boolean isDocTypeAccepted(String docType) {
+        List<String> types = getDocTypes();
+        return types == null || types.isEmpty() || types.contains(docType);
+    }
+
+    public boolean isEnabled() {
+        Boolean enabled = (Boolean) document.getPropertyValue(RetentionConstants.ENABLED_PROP);
+        return Boolean.TRUE.equals(enabled);
+    }
+
+    public boolean isEventBased() {
+        return getStartingPointPolicy() == StartingPointPolicy.EVENT_BASED;
+    }
+
+    public boolean isImmediate() {
+        return getStartingPointPolicy() == StartingPointPolicy.IMMEDIATE;
+    }
+
+    public boolean isManual() {
+        return ApplicationPolicy.MANUAL.name().toLowerCase().equals(getApplicationPolicy());
+    }
```

Comment:
can be replaced by `return ApplicationPolicy.MANUAL.name().equalsIgnoreCase(getApplicationPolicy());`